pypsa/statistics.py

```diff
@@ -559,10 +560,14 @@ def optimal_capacity(
 
         if storage:
             comps = ("Store", "StorageUnit")
+        if bus_carrier:
+            at_port = True
```

Comment:
This is a hard override of the keyword argument, which may lead to unexpected behaviour. We can instead put the default of `at_port` to None and only override if None. 
```python
    if bus_carrier and at_port is None:
            at_port = True
```

Reply:
Or perhaps even better: not overloading the `at_port` argument: Instead create a new keyword argument `absolute` / `absolute_capacities` which is None and aligns to `bus_carrier` per default.
```python
if bus_carrier and absolute_capacities is None:
    absolute_capacities = True
```

Reply:
The reason I used `at_port=True` is that it is more similar to how we had it before introducing the `at_port` parameter. At that time, we always looped over all ports to check if any bus had the provided `bus_carrier`. With the current implementation in the master, when selecting `bus_carrier="electricity"`, for example, you do not get links that have the `bus_carrier` "electricity" at port 2, e.g. CHP in PyPSA-Eur. Hence, it looks like they do not exist in the network.

Reply:
Your first suggestion is valid and I will adapt it.